homeassistant/components/template/fan.py

```diff
@@ -117,7 +184,45 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the template fans."""
-    async_add_entities(await _async_create_entities(hass, config))
+    if discovery_info is None:
+        async_add_entities(
+            await _async_create_entities(
+                hass,
+                rewrite_legacy_to_modern_conf(hass, config[CONF_FANS]),
+            )
+        )
+        return
+
+    async_add_entities(
+        await _async_create_entities(
+            hass,
+            discovery_info["entities"],
+            discovery_info["unique_id"],
+        )
+    )
```

Comment:
I think we should fail here if there's a coordinator in the discovery info instead of silently configuring a non-trigger based template fan